lib/include/tick/base_model/model_labels_features.h

```diff
@@ -73,15 +73,40 @@ class DLL_PUBLIC TModelLabelsFeatures : public virtual TModel<T> {
 
   template <class Archive>
   void save(Archive &ar) const {
+    ar(cereal::make_nvp("Model", cereal::base_class<TModel<T> >(this)));
     ar(CEREAL_NVP(n_samples));
+    ar(CEREAL_NVP(n_features));
+    ar(CEREAL_NVP(ready_columns_sparsity));
+    ar(CEREAL_NVP(column_sparsity));
+
     ar(cereal::make_nvp("labels", *labels));
     ar(cereal::make_nvp("features", *features));
   }
+
+ protected:
+  BoolStrReport compare(const TModelLabelsFeatures<T> &that,
+                        std::stringstream &ss) {
+    if (!this->features || !that.features)
+      TICK_ERROR("Cannot compare null features");
+    if (!this->labels || !that.labels) TICK_ERROR("Cannot compare null labels");
```

Comment:
Is there no way we can encompass this check in the array compare?

Reply:
not really, they're shared_ptrs which might be null - and thus we might have no reference to an array

Reply:
Ok, and can't `TICK_CMP_REPORT_PTR` check that if one of them is a nullptr then the other should be as well?